tests/foundry/client-esm/applications/ui/context.test-d.ts

```diff
@@ -0,0 +1,33 @@
+import { expectTypeOf } from "vitest";
+
+expectTypeOf();
+
+const myJournal = new JournalEntry.implementation();
+
+const testAppV1 = new JournalSheet(myJournal);
+const testAppV2 = new foundry.applications.sheets.journal.JournalEntrySheet({ document: myJournal });
+
+declare const htmlElement: HTMLElement;
+declare const jquery: JQuery;
+
+const menuA = foundry.applications.ui.ContextMenu.create(testAppV2, testAppV2.element, ".foobar", {
+  jQuery: false,
+  fixed: true,
+});
+expectTypeOf(menuA.fixed).toEqualTypeOf<true>();
+expectTypeOf(menuA.menuItems[0]?.callback(htmlElement)).toBeVoid();
+// @ts-expect-error Callback for jQuery: false takes an HTML Element
+expectTypeOf(menuA.menuItems[0]?.callback(jquery)).toBeVoid();
+
+// Deprecated menu items for appv2
+const menuB = foundry.applications.ui.ContextMenu.create(testAppV2, testAppV2.element, ".foobar", []);
+expectTypeOf(menuB.fixed).toEqualTypeOf<false>();
+expectTypeOf(menuB.menuItems[0]?.callback(jquery)).toBeVoid();
+// @ts-expect-error Callback for jQuery: true takes JQuery
+expectTypeOf(menuB.menuItems[0]?.callback(htmlElement)).toBeVoid();
+
+const menuC = foundry.applications.ui.ContextMenu.create(testAppV1, testAppV1.element, ".foobar", []);
+expectTypeOf(menuC.fixed).toEqualTypeOf<false>();
+expectTypeOf(menuC.menuItems[0]?.callback(jquery)).toBeVoid();
+// @ts-expect-error Callback for jQuery: true takes JQuery
+expectTypeOf(menuC.menuItems[0]?.callback(htmlElement)).toBeVoid();
```

Comment:
Sam here, when using `@ts-expect-error` remove as much as possible that could go wrong.